packages/store/src/workspace/page.ts

```diff
@@ -313,6 +313,37 @@ export class Page extends Space<PageData> {
     return parent.children.slice(index + 1);
   }
 
+  @debug('CRUD')
+  public addBlocksByFlavour<
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ALLProps extends Record<string, any> = BlockSuiteModelProps.ALL,
+    Flavour extends keyof ALLProps & string = keyof ALLProps & string
+  >(
+    blocks: Array<{
+      flavour: Flavour;
+      blockProps: Partial<
+        ALLProps[Flavour] &
+          Omit<BlockSuiteInternal.IBaseBlockProps, 'flavour' | 'id'>
+      >;
+    }>,
+    parent?: BaseBlockModel | string | null,
+    parentIndex?: number
+  ): string[] {
+    const ids: string[] = [];
+    blocks.forEach(block => {
+      const id = this.addBlockByFlavour<ALLProps, Flavour>(
+        block.flavour,
+        block.blockProps ?? {},
+        parent,
+        parentIndex
+      );
+      ids.push(id);
+      typeof parentIndex === 'number' && parentIndex++;
```

Comment:
If no `parentIndex` is specified, will the subsequent blocks always being added to the first position?

Reply:
If you are going to add new API in `Page`, I think the test case in `workspace.unit.spec.ts` should be provided altogether.

Reply:
>If you are going to add new API in Page, I think the test case in workspace.unit.spec.ts should be provided altogether.

working on it

Subsequent blocks will be added at the end page of top level (`const parentId = parent === null ? null : parent?.id ?? this.root?.id;`) in order (`const index = parentIndex ?? yChildren.length;`),

Reply:
Updated at https://github.com/toeverything/blocksuite/pull/1188/commits/111120e449eed470bcb1b70809f491ec55d5a637